diff --git a/dist/index.d.mts b/dist/index.d.mts
index c2e49f1cabd999dc7126dc72165e1553f0d65c06..5c743752f8556faa0e60bd0614eff0977442b5a1 100644
--- a/dist/index.d.mts
+++ b/dist/index.d.mts
@@ -1,14 +1,19 @@
 import { PlatformProxy } from 'wrangler';
 
+export interface Env {
+}
+
+type TypedPlatformProxy = PlatformProxy<Env>
+
 declare module "h3" {
     interface H3EventContext {
-        cf: PlatformProxy["cf"];
+        cf: TypedPlatformProxy["cf"];
         cloudflare: {
             request: Request & {
-                cf: PlatformProxy["cf"];
+                cf: TypedPlatformProxy["cf"];
             };
-            env: PlatformProxy["env"];
-            context: PlatformProxy["ctx"];
+            env: TypedPlatformProxy["env"];
+            context: TypedPlatformProxy["ctx"];
         };
     }
 }
@@ -25,3 +30,4 @@ declare module "nitropack" {
 declare const _default: (arg1: unknown, arg2: unknown) => void;
 
 export { _default as default };
+
diff --git a/dist/index.d.ts b/dist/index.d.ts
index c2e49f1cabd999dc7126dc72165e1553f0d65c06..77eda967078967ccd47b949bb12da58ac058e469 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,14 +1,19 @@
 import { PlatformProxy } from 'wrangler';
 
+export interface Env {
+}
+
+type TypedPlatformProxy = PlatformProxy<Env>
+
 declare module "h3" {
     interface H3EventContext {
-        cf: PlatformProxy["cf"];
+        cf: TypedPlatformProxy["cf"];
         cloudflare: {
             request: Request & {
-                cf: PlatformProxy["cf"];
+                cf: TypedPlatformProxy["cf"];
             };
-            env: PlatformProxy["env"];
-            context: PlatformProxy["ctx"];
+            env: TypedPlatformProxy["env"];
+            context: TypedPlatformProxy["ctx"];
         };
     }
 }
